scripts/release.js

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env node
+
+/* eslint-disable no-console */
+/* eslint-disable no-process-exit */
+/* eslint-disable import/no-commonjs */
+
+const { execSync } = require('child_process');
+const colors = require('colors/safe');
+const inquirer = require('inquirer');
+
+// check if user can publish new version to npm
+try {
+  execSync('$(npm owner add `npm whoami`)');
+} catch (e) {
+  console.log(
+    colors.red(`
+    You are not an owner of the npm repository,
+    ask for it before trying to bundle a release.
+  `)
+  );
+  process.exit(1);
+}
+
+// check if branch is clean with all changes commited
+const uncommitedChanges = execSync('git status --porcelain').toString().trim();
+if (uncommitedChanges) {
+  console.log(
+    colors.red(`
+    Working tree is not clean, please commit all your changes before release.
+  `)
+  );
+  process.exit(1);
+}
+
+const currentBranch = execSync('git rev-parse --abbrev-ref HEAD')
+  .toString()
+  .trim();
+
+// check if we are not on branch master or exit
+if (currentBranch === 'master') {
+  console.log(
+    colors.red(`
+    It's not possible to release a new version from the master branch.
+    Please checkout any other branch:
+      - develop: release new stable version
+      - xxxxxxx: release new beta version
+  `)
+  );
+  process.exit(1);
+}
+
+const strategy = currentBranch === 'develop' ? 'stable' : 'beta';
+
+// called if process aborted before publish,
+// nothing is pushed nor published remove local changes
+function rollback(newVersion) {
+  if (strategy === 'stable') {
+    // reset master
+    execSync('git reset --hard origin master');
+    execSync('git checkout develop');
+  } else {
+    // remove last commit
+    execSync('git reset --hard HEAD~1');
+  }
+
+  // remove local created tag
+  execSync(`git tag -d v${newVersion}`);
+  process.exit(1);
+}
+
+inquirer
+  .prompt([
+    {
+      type: 'confirm',
+      name: 'confirm',
+      message: colors.yellow.underline(
+        `You are on "${currentBranch}" branch. Are you sure you want to release a ${strategy ===
+        'stable'
+          ? 'new stable'
+          : 'beta'} version?`
+      ),
+      default: false,
+    },
+  ])
+  .then(({ confirm }) => {
+    if (!confirm) process.exit(0);
+
+    if (strategy === 'stable') {
+      // merge develop on master first
+      // and start release from master branch
+      console.log(colors.blue('Updating working tree'));
+      execSync('git checkout master');
+      execSync('git pull origin master');
+      execSync('git fetch origin --tags');
+      execSync('git fetch origin develop');
+      execSync('git merge origin/develop');
+    }
+
+    console.log(colors.blue('Install dependencies'));
+    execSync('yarn');
```

Comment:
The "safest" way is to:

```shell
rm -rf node_modules docgen/node_modules && 
    yarn cache clean &&
    yarn

cd docgen && yarn
```

This way we will have no cache, same packages as described into the `yarn.lock` and `package.json`

What do you think?